lib/iris/tests/unit/plot/test__get_plot_objects.py

```diff
@@ -0,0 +1,29 @@
+# Copyright Iris contributors
+#
+# This file is part of Iris and is released under the LGPL license.
+# See COPYING and COPYING.LESSER in the root of the repository for full
+# licensing details.
+"""Unit tests for the `iris.plot._get_plot_objects` function."""
+
+# Import iris.tests first so that some things can be initialised before
+# importing anything else.
+import iris.tests as tests  # isort:skip
+
+import iris.cube
+
+if tests.MPL_AVAILABLE:
+    from iris.plot import _get_plot_objects
+
+
+@tests.skip_plot
+class Test_get_plot_objects(tests.IrisTest):
```

Comment:
Given we're adding tests for `_get_plot_objects`, we could add test for the other modes it has too. Though those are presumably covered elsewhere so I don't mind.

Reply:
> presumably covered elsewhere

The fact that my initial change would have broken the error message rather suggests we don't have good coverage for this function.  I think I'd like to think about this more next week when I'm feeling more awake!

Reply:
Looks like [Test1dScatter](https://github.com/SciTools/iris/blob/35d97ed9bed0ccefa6d940d7d1d40a89b6ce0d24/lib/iris/tests/test_plot.py#L245) covers most of the eventualities.

Reply:
OK, I think I've convinced myself that all the various input permutations (one or two args, coords and/or cubes, different size inputs) for `_get_plot_objects` are exercised by [Test1dPlotMultiArgs](https://github.com/SciTools/iris/blob/35d97ed9bed0ccefa6d940d7d1d40a89b6ce0d24/lib/iris/tests/test_plot.py#L143) and [Test1dScatter](https://github.com/SciTools/iris/blob/35d97ed9bed0ccefa6d940d7d1d40a89b6ce0d24/lib/iris/tests/test_plot.py#L245).  The bit about plotting against a vertical coordinate was added quite recently (#3906) and, as well as updating some hashes for `Test1dPlotMultiArgs`, it [added specific quickplot tests](https://github.com/SciTools/iris/pull/3906/files#diff-66d543e25830127bbe1610fc11a967ab98604a3254f39ccb6851cb925dc0e94d).  So although it didn't have direct unit tests, I think we do have thorough coverage for this function.

This does make me wonder whether I've done the right thing in adding this module and should have added some tests for `plot` and/or `scatter` instead?

Reply:
I was tempted to suggest adding your new tests to Test1dScatter rather than making separate unit tests that aren't overly linked back to the main ones. I think I like unit testing this rather than just adding more integration tests (with images particularly) though, so happy with this choice.